connector/connection.go

```diff
@@ -60,6 +60,13 @@ type ConnectionListener interface {
 // NewMQTTConnection creates a local MQTT connection.
 func NewMQTTConnection(
 	config *Configuration, clientID string, logger watermill.LoggerAdapter,
+) (*MQTTConnection, error) {
+	return NewMQTTConnectionWithCredentials(config, clientID, logger, nil)
+}
+
+// NewMQTTConnectionWithCredentials creates a local MQTT connection with specific credentials provider.
```

Comment:
1. In Go, such "With" methods are not used at all. Here are a few examples:
   * https://pkg.go.dev/bufio#NewReaderSize
   * https://pkg.go.dev/bytes#NewBufferString
   * https://golang.org/pkg/compress/flate/#NewReaderDict
1. credentials func()... - these are not credentials but credentials provider, the parameter name should be fixed
1. The method documentation should be more complete i.e. to cover all parameters, to describe the case where credentials provider and the configuration overlap.